apis/externalsecrets/v1beta1/secretstore_validator.go

```diff
@@ -54,9 +55,26 @@ func (r *GenericStoreValidator) ValidateDelete(_ context.Context, _ runtime.Obje
 }
 
 func validateStore(store GenericStore) (admission.Warnings, error) {
+
+	if len(store.GetSpec().Conditions) != 0 {
+		validateConditions(store)
+	}
 	provider, err := GetProvider(store)
 	if err != nil {
 		return nil, err
 	}
 	return nil, provider.ValidateStore(store)
 }
+
+func validateConditions(store GenericStore) error {
+	for _, condition := range store.GetSpec().Conditions {
+		if len(condition.NamespacesRegex) != 0 {
+			for _, r := range condition.NamespacesRegex {
+				if _, err := regexp.Compile(r); err != nil {
+					return err
```

Comment:
Returning the error directly may not be very user-friendly. Could you wrap it with additional information like the one below? 

```
	for ri, condition := range store.GetSpec().Conditions {
		for ci, r := range condition.NamespacesRegex {
			if _, err := regexp.Compile(r); err != nil {
				return fmt.Errorf("failed to compile %dth namespace regex in %dth condition: %w", ri, ci, err)
			}
		}
	}
```